src/current/v25.2/changefeed-message-envelopes.md

```diff
@@ -0,0 +1,795 @@
+
+In CDC queries:
+
+A changefeed containing a `SELECT` clause without any additional options:
+
+~~~sql
+CREATE CHANGEFEED INTO 'external://kafka' AS SELECT city, type FROM movr.vehicles;
+~~~
+~~~
+{"city": "los angeles", "type": "skateboard"}
+~~~
+
+A changefeed containing a `SELECT` clause with the [`topic_in_value`]({% link {{ page.version.version }}/create-changefeed.md %}#topic-in-value) option specified:
+
+~~~sql
+CREATE CHANGEFEED INTO 'external://kafka' WITH topic_in_value AS SELECT city, type FROM movr.vehicles;
+~~~
+~~~
+{"__crdb__": {"topic": "vehicles"}, "city": "los angeles", "type": "skateboard"}
+~~~
+
+#### `key_only`
+
+`key_only` emits only the key and no value, which is faster if you only need to know the key of the changed row. This envelope option is only supported for [Kafka sinks]({% link {{ page.version.version }}/changefeed-sinks.md %}#kafka) or sinkless changefeeds.
+
+Kafka sink:
+
+~~~sql
+CREATE CHANGEFEED FOR TABLE users INTO 'external://kafka' WITH envelope=key_only;
+~~~
+~~~
+["boston", "22222222-2222-4200-8000-000000000002"]
+~~~
+
+{{site.data.alerts.callout_info}}
+It is necessary to set up a [Kafka consumer](https://docs.confluent.io/platform/current/clients/consumer.html) to display the key because the key is part of the metadata in Kafka messages, rather than in its own field. When you start a Kafka consumer, you can use `--property print.key=true` to have the key print in the changefeed message.
+{{site.data.alerts.end}}
+
+Sinkless changefeeds:
+
+~~~sql
+CREATE CHANGEFEED FOR TABLE users WITH envelope=key_only;
+~~~
+~~~
+{"key":"[\"seattle\", \"fff726cc-13b3-475f-ad92-a21cafee5d3f\"]","table":"users","value":""}
+~~~
+
+#### `row`
+
+`row` emits the row without any additional metadata fields in the message. This envelope option is only supported for [Kafka sinks]({% link {{ page.version.version }}/changefeed-sinks.md %}#kafka) or sinkless changefeeds. `row` does not support [`avro`]({% link {{ page.version.version }}/changefeed-messages.md %}#avro) format—if you are using `avro`, refer to the [`bare`](#bare) envelope option.
+
+Kafka sink:
+
+~~~sql
+CREATE CHANGEFEED FOR TABLE vehicles INTO 'external://kafka' WITH envelope=row;
+~~~
+~~~
+{"city": "washington dc", "creation_time": "2019-01-02T03:04:05", "current_location": "85551 Moore Mountains Apt. 47", "ext": {"color": "red"}, "id": "d3b37607-1e9f-4e25-b772-efb9374b08e3", "owner_id": "4f26b516-f13f-4136-83e1-2ea1ae151c20", "status": "available", "type": "skateboard"}
+~~~
+
+## Field reference
+
+CockroachDB provides multiple changefeed envelopes, each supported by different [sinks]({% link {{ page.version.version }}/changefeed-sinks.md %}) and [use cases](#use-cases).
+
+### `"payload"`
+
+The change event data. `"payload"` is a wrapper only with [webhook]({% link {{ page.version.version }}/changefeed-sinks.md %}#webhook-sink) sinks and when the [`enriched_properties`](#enriched-properties-option) options is used.
```

Comment:
i guess there are two different payloads in question here, which is a little confusing. can you try to disambiguate them?

Reply:
maybe this organization/format doesnt make that much sense given that we're mixing fields from various envelopes together. this makes it seem like you can get all these fields but actually you only get a subset depending on envelope

Reply:
Added bullet points to the `payload` section to call out the difference between the two types of `payload` wrappers mentioned here. Lmk if you wanted something else.

As a reference, I think this works fine, and each description comes with the option/configuration that enables that field. The alternative would involve every combination as an example that imo would be harder to reference. Given the large number of examples already on this page, I'd like to give this format a shot, but be ready to adapt if we get signal this isn't working.

Reply:
I also think the fields and options are kind of ripe for automation something like what docs do for the cluster settings page; however, that is a much bigger project for another day involving more stakeholders.